assets/prototype/domain/eventEditor/tickets/ticketList/filterBar/useTicketListFilterState/index.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import { useReducer, useEffect } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import filters from '../../../../../shared/predicates/tickets/filters';
+import { DisplayDates, ShowTickets, TicketsSortedBy } from '../../../../data/ticket/types';
+import { Ticket } from '../../../../data/types';
+
+enum ActionType {
+	SET_DISPLAY_TICKET_DATE = 'SET_DISPLAY_TICKET_DATE',
+	SET_SHOW_TICKETS = 'SET_SHOW_TICKETS',
+	SET_TICKETS_SORTED_BY = 'SET_TICKETS_SORTED_BY',
+	TOGGLE_IS_CHAINED = 'TOGGLE_IS_CHAINED',
+}
+
+interface TicketListFilterState {
+	displayTicketDate: DisplayDates;
+	isChained: boolean;
+	processedTickets: Ticket[];
+	showTickets: ShowTickets;
+	tickets: Ticket[];
+	ticketsSortedBy: TicketsSortedBy;
+}
+
+const useTicketListFilterState = (tickets: Ticket[]): TicketListFilterState => {
+	const initialState = {
+		displayTicketDate: DisplayDates.start,
+		isChained: false,
+		processedTickets: [],
+		showTickets: ShowTickets.all,
+		tickets,
+		ticketsSortedBy: TicketsSortedBy.chronologically,
+	};
+	const [state, dispatch] = useReducer(reducer, initialState);
+
+	useEffect(() => {
+		// console.log('useTicketListFilterState >>>', state);
+	}, [state]);
+
+	const setDisplayTicketDate = (displayTicketDate) => {
+		dispatch({
+			type: ActionType.SET_DISPLAY_TICKET_DATE,
+			displayTicketDate,
+		});
+	};
+
+	const setShowTickets = (showTickets) => {
+		dispatch({
+			type: ActionType.SET_SHOW_TICKETS,
+			showTickets,
+		});
+	};
+
+	const setTicketsSortedBy = (ticketsSortedBy) => {
+		dispatch({
+			ticketsSortedBy,
+			type: ActionType.SET_TICKETS_SORTED_BY,
+		});
+	};
+
+	const toggleIsChained = () => {
+		dispatch({
+			type: ActionType.TOGGLE_IS_CHAINED,
+		});
+	};
+
+	return {
+		...state,
+		setDisplayTicketDate,
+		setShowTickets,
+		setTicketsSortedBy,
+		toggleIsChained,
+	};
+};
+
+interface Action {
+	type: ActionType;
+}
+
+const reducer = (state, action) => {
```

Comment:
Any reasons for not typefying it or that is going to be a separate PR?

Reply:
This PR got too hairy.
I will come back on this area when I will couple to UI and will do some refactoring and types also will be added.